src/transport.ts

```diff
@@ -58,52 +58,39 @@ export class WebRTCTransport implements Transport, Initializable {
       throw inappropriateMultiaddr("we need to have the remote's PeerId");
     }
 
-    let peerConnection = new RTCPeerConnection();
+    let certificate = await RTCPeerConnection.generateCertificate({
+      name: 'ECDSA',
+      namedCurve: 'P-256',
+    } as any);
+    let peerConnection = new RTCPeerConnection({ certificates: [certificate] });
 
     // create data channel
+    let dataChannelOpenPromise = defer();
     let handshakeDataChannel = peerConnection.createDataChannel('data', { negotiated: true, id: 1 });
-    //
+    handshakeDataChannel.onopen = (_) => dataChannelOpenPromise.resolve();
+    handshakeDataChannel.onerror = (ev: Event) => {
+      log.error('Error opening a data channel for handshaking: %s', ev.toString());
+      dataChannelOpenPromise.reject(dataChannelError('data', `error opening datachannel: ${ev.toString()}`));
+    };
+    setTimeout(() => {
```

Comment:
you never cancel this time out as you do not assign it. So while the promise won't be affected, you will still log the error?!

Reply:
good catch! thank you